aviatrix_ha/__init__.py

```diff
@@ -54,23 +56,17 @@ def _lambda_handler(event, context):
         print("From SNS Event")
     except (AttributeError, IndexError, KeyError, TypeError):
         pass
-    if os.environ.get("TESTPY") == "True":
-        print("Testing")
-        client = boto3.client(
-            "ec2",
-            region_name=os.environ["AWS_TEST_REGION"],
-            aws_access_key_id=os.environ["AWS_ACCESS_KEY_BACK"],
-            aws_secret_access_key=os.environ["AWS_SECRET_KEY_BACK"],
-        )
-        lambda_client = boto3.client(
-            "lambda",
-            region_name=os.environ["AWS_TEST_REGION"],
-            aws_access_key_id=os.environ["AWS_ACCESS_KEY_BACK"],
-            aws_secret_access_key=os.environ["AWS_SECRET_KEY_BACK"],
-        )
-    else:
-        client = boto3.client("ec2")
-        lambda_client = boto3.client("lambda")
+    try:
+        function_headers = event["headers"]
+        function_request = event["requestContext"]
+        print("From Function Request")
+    except (KeyError, AttributeError):
+        function_headers = None
+        function_request = None
+        print("Not from Function Request")
+
+    client = boto3.client("ec2")
+    lambda_client = boto3.client("lambda")
```

Comment:
Could we move lines 46 through 66 to its own function? This will remove some of the exception handling and seems a bit simpler and just return the type as a string

```

def get_event_type(event):
    if "StackId" in event:
        return "cloudformation"
    if "Records" in event and event["Records"]:
        if event["Records"][0].get("EventSource") == "aws:sns":
            return "sns"
    if "headers" in event and "requestContext" in event:
        return "function_request"
    return "unknown"
```

Then do

```
event_type = get_event_type(event)
....
print(f"[INFO] Version: {VERSION}, Event: {event}")
client = boto3.client("ec2")
lambda_client = boto3.client("lambda")
....

And then below we do this:

```
    if event_type == "cloudformation":
        return handle_cft(
            describe_err,
            event,
            context,
            client,
            lambda_client,
            controller_instanceobj,
            instance_name,
        )
    elif event_type == "sns":
        return handle_sns_event(
            describe_err, event, client, lambda_client, controller_instanceobj, context
        )
    elif event_type == "function_request":
        return handle_function_event(event, context)
    else:
        print("Unknown source. Not from CFT, SNS, or Function Request.")
        return False
```

Reply:
done